### PR TITLE
create protocol adapter

### PR DESCRIPTION
**Pull request by colombod**
The goal is to strengthen the transformation between domain entities and jupyter protocol
- addresses also #437

### CODE REVIEW

File: Microsoft.DotNet.Interactive.Jupyter/IPubSubChannel.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.Interactive.Jupyter.Protocol;
+
+namespace Microsoft.DotNet.Interactive.Jupyter
+{
+    public interface IPubSubChannel
+    {
+        bool Send(JupyterPubSubContent content, Message request, string ident = null);
+    }
```
Comment (jonsequitur) [inline Microsoft.DotNet.Interactive.Jupyter/IPubSubChannel.cs]: Why are content and request different?

Comment (jonsequitur) [inline Microsoft.DotNet.Interactive.Jupyter/IPubSubChannel.cs]: Why the bool return type? What does it signify? Also, should this be async?

Comment (jonsequitur) [inline Microsoft.DotNet.Interactive.Jupyter/IPubSubChannel.cs]: What's the use case and expectation for ident being null?

File: Microsoft.DotNet.Interactive.Jupyter/ReplyChannel.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        }
+        public bool Send(JupyterReplyContent content, Message request)
+        {
+            var reply = Message.CreateReply(content, request);
+            return _sender.Send(reply);
+        }
+    }
+
+    public class PubSubChannel : IPubSubChannel
+    {
```
Comment (jonsequitur) [inline Microsoft.DotNet.Interactive.Jupyter/ReplyChannel.cs]: Separate file please.

File: Microsoft.DotNet.Interactive.Jupyter/IPubSubChannel.cs (hunk 1 above)
Comment (colombod) [reply Microsoft.DotNet.Interactive.Jupyter/IPubSubChannel.cs]: not always needed to generate the topic

**Pull request merged**